genai_stack/memory/vectordb.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional, Dict
+from langchain.memory import VectorStoreRetrieverMemory
+from genai_stack.memory.base import BaseMemory, BaseMemoryConfig, BaseMemoryConfigModel
+from genai_stack.memory.utils import parse_chat_conversation_history_search_result, extract_text, format_index_name
+
+
+class VectorDBMemoryConfigModel(BaseMemoryConfigModel):
+    """Data Model for the configs"""
+
+    retrieve_parameters:Optional[Dict] = {
+        'search_type':'similarity',
+        'search_kwargs':{'k':4}
+    }
+    chromadb:Optional[Dict] = {
+        'index_name':'Test_history'
+    }
+    weaviate:Optional[Dict] = {
```

Comment:
I dont think we should be having separate configs for each db . If a db needs an extra_kwarg like text_key we can hardcode that value or generate that value based on the index_name. index_name should be the only config for a memory component in my opinion.